src/libraries/Microsoft.Extensions.Logging.Abstractions/ref/Microsoft.Extensions.Logging.Abstractions.csproj

```diff
@@ -1,9 +1,12 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;net461</TargetFrameworks>
```

Comment:
CC @maryamariyan @eerhardt 
You're adding either `NETCoreAppCurrent` or `netcoreapp3.1` configurations to a lot of libraries.

What made you choose one vs another?  For those that are `NETCoreAppCurrent` do you expect to add more configurations with every release?

Reply:
In this specific case I added the `NetCoreAppCurrent` tfm to the reference project as it was already present in the source project. My strategy was:

1. Make sure that the assembly universe matches between the reference and the source assemblies to avoid an unnecessary reference to the netstandard.dll shim. This is a requirement inside the shared framework but is also desirable in OOB projects to avoid the shim altogether.
2. In cases where multiple .NETCoreApp versions are present in the source project, choose the minimum required .NETCoreApp tfm in the reference project which in most cases is `netcoreapp3.1`.
3. Add a `NetCoreAppCurrent` tfm to OOBs which don't target .NETCoreApp but where it's desirable to avoid the netstandard.dll shim. The desire to avoid the netstandard.dll shim is defined by where the library sits in the graph. If it's at the very top and isn't referenced by any other component it's less of a concern.

> For those that are NETCoreAppCurrent do you expect to add more configurations with every release?

The `netcoreapp3.1` tfms will be upgraded to `net6.0` and the `NetCoreAppCurrent` tfms will be replaced by `net6.0` + `NetCoreAppCurrent` when the repository targets .NET 7.

Related but not yet tackled, there's the discussion to add a .NETCoreApp (presumably `NetCoreAppCurrent`) tfm to every  applicable library that already targets .NETStandard to benefit from linker/compiler/analyzer features. I would like to keep that discussion separate from this effort though.

Reply:
> to benefit from linker/compiler/analyzer features

Also to benefit from new APIs as well.

I'm all for targeting the latest TFM in our OOBs all the time. Especially in the OOBs that go inbox in other teams shared frameworks. I like that we have a decent plan so the TFMs don't continually grow and this strategy will be maintainable.